src/test/ui/extern/extern-crate-visibility.stderr

```diff
@@ -1,14 +1,26 @@
 error[E0603]: crate `core` is private
   --> $DIR/extern-crate-visibility.rs:6:10
    |
+LL |     extern crate core;
+   |     ------------------
+   |     |
+   |     used restricted re-export of `core`
```

Comment:
I think it's unclear what this message means. Perhaps something like: "`core` is declared in this restricted re-export", though I'm not sure "restricted re-export" is clear, especially to a beginner.

Reply:
The reversal of this message and the `this crate is private` message would really help here too.

Reply:
This is an unintended effect of this PR. I didn't bother to change deal with it because it is 2015, but it can be dramatically improved.

Reply:
Small improvement, we now only point to this once, instead of twice.